OsmAnd/src/net/osmand/plus/plugins/monitoring/MonitoringSettingsFragment.java

```diff
@@ -531,6 +539,11 @@ public void onDisplayPreferenceDialog(Preference preference) {
 				ApplicationMode appMode = getSelectedAppMode();
 				SingleSelectPreferenceBottomSheet.showInstance(fm, prefId, this, false, appMode, true, true);
 			}
+		} else if (prefId.equals(RECORD_OBD_DATA)) {
+			MapActivity mapActivity = getMapActivity();
+			if (mapActivity != null) {
+				VehicleMetricsRecordingFragment.showInstance(mapActivity, this);
```

Comment:
Also, we could set fragment in prefs xml, similar to [CoordinatesFormatFragment](https://github.com/osmandapp/Osmand/blob/86b69b187906d896db4ffc8f43c0f847b917431a/OsmAnd/res/xml/general_profile_settings.xml#L57)